filebeat/prospector/prospector.go

```diff
@@ -36,11 +36,15 @@ type Prospectorer interface {
 	Run()
 }
 
-func NewProspector(cfg *common.Config, states file.States, spoolerChan chan *input.Event) (*Prospector, error) {
+type Outlet interface {
+	OnEvent(event *input.Event) bool
```

Comment:
I used `OnEvent` to indicate this being kind of a callback. `Send` sounds too much like a channel/io-whatever too me.